examples/joule_thief/joule_thief.py

```diff
@@ -0,0 +1,91 @@
+from pycircuit.device import Device, Pin
+from pycircuit.circuit import circuit, Node, Ref, Net
+from pycircuit.footprint import Footprint, Map
+from pycircuit.package import Package, RectCrtyd, DualPads
+from pycircuit.pcb import Pcb
+from pycircuit.formats import * #extends Pcb and Node
+from pycircuit.library.connectors import * #defines some Devices
+from pykicad.pcb import Zone #extends Pcb
+
+
+Footprint('R0805', 'R', '0805',
+          Map(1, '1'),
+          Map(2, '2'))
+
+
+Footprint('D0805', 'D', '0805',
+          Map(1, 'A'),
+          Map(2, 'K'))
+
+
+Footprint('BAT0805', 'DCCONN', '0805',
+          Map(1, 'V'),
+          Map(2, 'GND'))
+
+
+for a, b, c in ('BCE', 'BEC', 'CBE', 'CEB', 'ECB', 'EBC'):
+    Footprint('SOT23' + a + b + c, 'Q', 'SOT23',
+              Map(1, a),
+              Map(2, b),
+              Map(3, c))
+
+
+Device('Transformer_1P_1S', pins=(Pin('1'), Pin('2'), Pin('3'), Pin('4')))
```

Comment:
I'd prefer the pins to be named like this:
Pin('L1.1', 'L1'), Pin('L1.2', 'L1'), Pin('L2.1', 'L2'), Pin('L2.2', 'L2')

'L1' and 'L2' are functions and mean that pin 1 and 2 are interchangeable